### PR TITLE
Fix recycled containers deleting items inside them

### PR DESCRIPTION
**Pull request by SlamBamActionman**
## About the PR

Right now you can delete any item that fits in a survival box by placing it inside of one and then recycling the box in a Recycler/Material Reclaimer. This PR fixes this by dumping out any item inside containers.

- For Reclaimers,  any recyclable items inside the container get placed on top of the Reclaimer.
- For Recyclers, any recyclable items inside the container get recursively recycled as well.

## Why / Balance

I doubt it's intended behavior to be able to effectively delete any item that fits in a box. Also ensures a recycled box provides the correct amount of material as its contents also get recycled.

## Technical details

When trying to recycle, once all checks are passed it iterates through the object's containers if it has any and dumps out any items onto the machine doing the recycling.

## Media

image
The result of recycling a survival box in a Recycler/Reclaimer.

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- fix: Recyclers no longer delete items stored inside of recycled entities.

### CODE REVIEW

**Comment by HelpTheSteak** [comment]
How am I gonna destroy all my contraband after I complete my objectives as thief now

**Comment by SlamBamActionman** [comment]
> How am I gonna destroy all my contraband after I complete my objectives as thief now

Ngl I've done the same. It just feels like a kinda jank and unintended way to get rid of evidence, plus it provides little to no interesting gameplay for either side.

**Comment by deltanedas** [comment]
> How am I gonna destroy all my contraband after I complete my objectives as thief now

tell the chaplain you found 4 among pequeno and you must personally incinerate them

**Comment by HelpTheSteak** [comment]
> How am I gonna destroy all my contraband after I complete my objectives as thief now

But anyway I think this is actually a good change since det has more evidence to check now (unless they cremated it)

File: Content.Shared/Materials/SharedMaterialReclaimerSystem.cs [hunk 1]
```diff
@@ -110,6 +111,21 @@ public bool TryStartProcessItem(EntityUid uid, EntityUid item, MaterialReclaimer
             component.NextSound = Timing.CurTime + component.SoundCooldown;
         }
 
+        if (TryComp<ContainerManagerComponent>(item, out var contManager)){
+            foreach (var cont in contManager.Containers.Values)
+            {
+                var containedEntites = cont.ContainedEntities.ToArray();
```
Comment (ElectroJr) [inline Content.Shared/Materials/SharedMaterialReclaimerSystem.cs:117]: This shouldn't iterate over all containers, it has to be opt-in. Not all containers are there to store physical entities that can be dropped. E.g., "Actions" in the actions hotbar are also entities in containers.

Instead, this should behave more like entity destruction/breaking. I.e., raise an event, then let the storage & entity-storage systems handle that event.

**Review by ElectroJr — CHANGES_REQUESTED** [review]

**Comment by SlamBamActionman** [comment]
This is the first time I'm making something new for the event system so let me know if there's something iffy with the implementation. Behavior seems to be replicated and working fine though.

File: Content.Shared/Materials/SharedMaterialReclaimerSystem.cs [hunk 2]
```diff
@@ -110,6 +112,9 @@ public bool TryStartProcessItem(EntityUid uid, EntityUid item, MaterialReclaimer
             component.NextSound = Timing.CurTime + component.SoundCooldown;
         }
 
+        var reclaimedEvent = new GotReclaimedEvent(uid.ToCoordinates());
+        RaiseLocalEvent(item, ref reclaimedEvent);
```
Comment (metalgearsloth) [inline Content.Shared/Materials/SharedMaterialReclaimerSystem.cs:116]: use xformsystem, these methods do a bunch of resolves internally that are slow and not thread-safe.

**Pull request merged**